### PR TITLE
Feature/qa/milestone/135

### PR DESCRIPTION
**Pull request by anirudhmt**

### CODE REVIEW

File: src/app/actions.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { post, get, put, del } from "@utils/apiUtils";
 import apiConfig from "@config/api.config";
 import commentModel from "@models/comment";
+// import transcriptionModel from "@models/transcription";
+import transcriptionUtils from "@utils/transcriptionUtils";
```
Comment (afrozMT) [inline src/app/actions.js]: Make it a model

File: src/app/actions.js [hunk 2]
```diff
@@ -444,5 +446,151 @@ export let actions = () => ({
...
+				...state.transcriptionPane,
+				searchBar: {
+					...state.transcriptionPane.searchBar,
+					currentMatchNumber: currentMatchNumber,
+				}
+			}
+		}
+	},
+
+	updateTranscriptionFilters: (state, { selectedEvalParams }) => {
```
Comment (afrozMT) [inline src/app/actions.js]: merge this with updateTranscriptionSearchWords method

File: src/app/components/MultiSelectDropDown/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+	isOptionSelected(option) {
+		return this.state.selectedOptions.some(selectedOption => option.value === selectedOption.value);
+	}
+
+	render() {
+		let options = this.props.options && Array.isArray(this.props.options) ? this.props.options : [];
+		return (
+		<div className={style.dropdownMenu}>
+			<ul style={{maxHeight: '150px'}}>
```
Comment (afrozMT) [inline src/app/components/MultiSelectDropDown/index.js]: move to scss

File: src/app/components/SearchBar/index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	handleKeyPress(event) {
+		if (event.key == 'Enter') {
+			this.addSearchWord(event.target.value);
+		}
+	}
+
+	addSearchWord(newWord) {
+		if (this.state.searchWords.includes(newWord)) {
```
Comment (afrozMT) [inline src/app/components/SearchBar/index.js]: use indexOf

File: src/app/components/SearchBar/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			this.setState( {inputValue: ""} );
+		}
+		else {
```
Comment (afrozMT) [inline src/app/components/SearchBar/index.js]: Can we remove if else ?

File: src/app/components/SearchBar/index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+						</div>
+						<div className={style.icon} onClick={() => this.removeSearchWord(searchWord)}>
+							<img src={closeImage} style="height:8px;" />
+						</div>
+						<div className={style.clear} />
+					</div>
+				))}
+
+				<input type="text" name="search" value={this.state.inputValue} onKeyPress={this.handleKeyPress}
+					placeholder="Search words in the transcription"  className={style.inputStyle} />
```
Comment (afrozMT) [inline src/app/components/SearchBar/index.js]: pass placeholder label as props

File: src/app/components/SearchNavigationBar/index.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import { h, Component } from "preact";
+import style from "./index.scss";
+class SearchNavigationBar extends Component {
+	constructor(props) {
+		super(props);
+		this.onUpClicked = this.onUpClicked.bind(this);
+		this.onDownClicked = this.onDownClicked.bind(this);
+		this.state = {
+			currentMatchNumber: 1
```
Comment (afrozMT) [inline src/app/components/SearchNavigationBar/index.js:10]: check if currentMatchNumber is needed

File: src/app/components/SearchNavigationBar/index.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+			this.setState({ currentMatchNumber: this.state.currentMatchNumber - 1 });
+		}
+		this.props.navigateToMatchHandler(this.state.currentMatchNumber)
+	}
+
+	render() {
+		return (
+			<div className={style.searchNavBar}>
+				<div className={style.resultRelatedText}>
+					<div style={this.props.numberOfMatches > 0 ? { display: 'none' } : null}>No match found</div>
```
Comment (afrozMT) [inline src/app/components/SearchNavigationBar/index.js]: pass it as props.

File: src/app/containers/FilterContainer/index.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+	hideFilters() {
+		this.setState({ showFilters: false });
+	}
+
+	render() {
+		let dropdownOptions = this.getOptionsFromEvalParams(this.props.evalParams);
+		let selectedOptions = this.getOptionsFromEvalParams(this.props.selectedEvalParams);
+		return (
+			<div className={style.selectFilter} onBlur={this.hideFilters}>
```
Comment (afrozMT) [inline src/app/containers/FilterContainer/index.js]: define a tabIndex={0} for blur to work

File: src/app/containers/FilterContainer/index.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+				<div className={style.filterIconHolder} onClick={this.onFilterClickHandler}>
+					<img src={filterImage} style="height:15px;" />
+				</div>
+				<div style={{ ...!this.state.showFilters ? { display: 'none' } : { display: 'block' }, width: '300px' }}>
```
Comment (afrozMT) [inline src/app/containers/FilterContainer/index.js]: remove all inline display styles. Add css class hide and show

File: src/app/containers/FilterContainer/index.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+					/>
+				</div>
+			</div>
+		);
+	}
+}
+
+
+function mapStateToProps(state) {
+	return {
```
Comment (afrozMT) [inline src/app/containers/FilterContainer/index.js]: recheck if evalParams and selectedEvalParams can be merged?

File: src/app/containers/SearchContainer/index.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+	}
+}
+
+function mapStateToProps(state) {
+	return {
+		searchWords: state.transcriptionPane.searchBar.searchWords,
+		numberOfMatches: state.transcriptionPane.searchBar.numberOfMatches,
+		currentMatchNumber: state.transcriptionPane.searchBar.currentMatchNumber,
+		selectedEvalParams: state.transcriptionPane.filter.selectedEvalParams,
+		timestampedTranscripts: state.transcriptionPane.timestampedTranscripts
```
Comment (afrozMT) [inline src/app/containers/SearchContainer/index.js]: Check if it is unused ?

File: src/app/containers/TranscriptionContainer/index.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	componentDidMount() {
+		this.props.getTimestampedTranscripts();
+	}
+
+	componentWillReceiveProps(nextProps) {
+		if (nextProps.currentMatchNumber != this.props.currentMatchNumber && nextProps.currentMatchNumber > 0) {
+			let transcriptIndex = nextProps.matchedTranscriptIndices[nextProps.currentMatchNumber - 1];
+			let elementId = "#commentCardId_" + this.props.searchedTranscripts[transcriptIndex].id;
```
Comment (afrozMT) [inline src/app/containers/TranscriptionContainer/index.js]: can we pass it from props.

File: src/app/styles/index.scss [hunk 14]
```diff
@@ -1,5 +1,7 @@
 @import "./colors.scss";
-
+body{
```
Comment (afrozMT) [inline src/app/styles/index.scss:2]: remove this font-family

**Review by afrozMT — APPROVED** [review]

File: src/app/actions.js (hunk 1 above)
Comment (anirudhmt) [reply src/app/actions.js]: Done

File: src/app/actions.js (hunk 2 above)
Comment (anirudhmt) [reply src/app/actions.js]: Done

File: src/app/components/SearchBar/index.js (hunk 4 above)
Comment (anirudhmt) [reply src/app/components/SearchBar/index.js]: done

File: src/app/components/SearchBar/index.js (hunk 5 above)
Comment (anirudhmt) [reply src/app/components/SearchBar/index.js]: removed

File: src/app/components/SearchBar/index.js (hunk 6 above)
Comment (anirudhmt) [reply src/app/components/SearchBar/index.js]: done

File: src/app/components/SearchNavigationBar/index.js (hunk 7 above)
Comment (anirudhmt) [reply src/app/components/SearchNavigationBar/index.js:10]: will revisit later

File: src/app/containers/FilterContainer/index.js (hunk 10 above)
Comment (anirudhmt) [reply src/app/containers/FilterContainer/index.js]: done

File: src/app/containers/FilterContainer/index.js (hunk 9 above)
Comment (anirudhmt) [reply src/app/containers/FilterContainer/index.js]: done

File: src/app/containers/FilterContainer/index.js (hunk 11 above)
Comment (anirudhmt) [reply src/app/containers/FilterContainer/index.js]: not doing now

File: src/app/containers/SearchContainer/index.js (hunk 12 above)
Comment (anirudhmt) [reply src/app/containers/SearchContainer/index.js]: done

File: src/app/styles/index.scss (hunk 14 above)
Comment (anirudhmt) [reply src/app/styles/index.scss:2]: done

File: src/app/components/SearchNavigationBar/index.js (hunk 8 above)
Comment (anirudhmt) [reply src/app/components/SearchNavigationBar/index.js]: done

File: src/app/containers/TranscriptionContainer/index.js (hunk 13 above)
Comment (anirudhmt) [reply src/app/containers/TranscriptionContainer/index.js]: done

File: package.json [hunk 15]
```diff
@@ -1,14 +1,14 @@
...
+  "version": "1.1.0",
   "description": "",
   "main": "dist/index.build.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "build:dev": "ENV=dev webpack --watch",
     "clean": "rm -rf dist",
     "build:prod-react": "ENV=prod BUILD_TYPE=react-build webpack",
-    "preversion": "git checkout master && git pull",
+    "preversion": "git checkout feature/qa/milestone/135 && git pull",
```
Comment (afrozMT) [inline package.json]: make sure this change is reverted .

**Pull request merged**